server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -63,6 +64,9 @@ public class TranslogTransferManager {
     private final RemoteTranslogTransferTracker remoteTranslogTransferTracker;
     private final RemoteStoreSettings remoteStoreSettings;
     private static final int METADATA_FILES_TO_FETCH = 10;
+    final static String CHECKPOINT_FILE_DATA_KEY = "ckp-data";
+    private final boolean shouldUploadTranslogCkpAsMetadata;
+    private final TranslogCheckpointSnapshotTransferManager transferManager;
```

Comment:
Not the right way, I have left a comment earlier on this class needs to be abstract and have implementors for the abstract method.